src/hotspot/share/gc/parallel/psPromotionManager.inline.hpp

```diff
@@ -284,6 +285,12 @@ inline oop PSPromotionManager::copy_unmarked_to_survivor_space(oop o,
     } else {
       // we'll just push its contents
       push_contents(new_obj);
+
+      if (psStringDedup::is_candidate_from_evacuation(o->klass(), new_obj->age(), new_obj_is_tenured)) {
+        if (!java_lang_String::test_and_set_deduplication_requested(o)) {
```

Comment:
Setting the flag on the from-object isn't useful; it needs to be on the to-object.  But I don't think this is needed anyway.  `is_candidate_from_evacuation` shouldn't generate duplicate requests.